sphinx_doc/tutorials/docs/writing_new_nav2planner_plugin.rst

```diff
@@ -0,0 +1,196 @@
+|                      | method should clean up resoures which are created for the planner.         |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+| createPlan()         | Method is called when planner server demands a global plan for specified   | Yes                     |
+|                      | start and goal pose. This method returns `nav_msgs::msg::Path` carrying    |                         |
+|                      | global plan. This method takes 2 input parmas: start pose and goal pose.   |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+
+For this tutorial, we will be using methods ``RRTConnect::configure()`` and ``RRTConnect::createPlan()`` to create RRT Connect planner.
+
+In our planner, ``RRTConnect::configure()`` method sets member variables and ROS parameters,
+
+.. code-block:: c
+
+  node_ = parent;
+  tf_ = tf;
+  name_ = name;
+  costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  // Parameter initialization
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".solve_time", rclcpp::ParameterValue(1.0));
+  node_->get_parameter(name_ + ".solve_time", solve_time_);
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".max_trials", rclcpp::ParameterValue(4));
+  node_->get_parameter(name_ + ".max_trials", max_trials_);
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".collision_checking_resolution", rclcpp::ParameterValue(0.001));
+  node_->get_parameter(name_ + ".collision_checking_resolution", collision_checking_resolution_);
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".allow_unknown", rclcpp::ParameterValue(true));
+  node_->get_parameter(name_ + ".allow_unknown", allow_unknown_);
+
+and also, it sets OMPL variables such as bounds, setup settings and space validation.
+
+In ``RRTConnect::createPlan()`` method, RRT Connect planner is called using start pose and goal pose to solve the global path planning problem. Upon succeding, it converts the path to the ``nav_msgs::msg::Path`` and returns to the caller which is planner server instance.
+
+The remaining methods are not used but its mandatory to override them. As per the rules, we did override all but left them blank.
+
+2- Exporting the planner plugin
+-------------------------------
+
+Now that we have created our custom planner, we need to export our planner plugin so that it would be visible to the planner server. Plugins are loaded at runtime and if they are not visible, then our planner server won't be able to load it. In ROS2, exporting and loading plugins is handled by ``pluginlib``.
+
+Coming to our tutorial, class ``nav2_rrtconnect_planner::RRTConnect`` is loaded dynamically as ``nav2_core::GlobalPlanner`` which is our base class.
```

Comment:
I'm not sure what you mean by loaded dynamically? at this point, you have some code in a file but there's no server. I think you just mean to point out that the planner derives from the GlobalPlanner class as a plugin interface.

Reply:
Yeah. I think this line is redundant. I should remove it.